src/tests-rosetta-cli-construction/validate-construction.ts

```diff
@@ -0,0 +1,182 @@
+import * as fs from 'fs';
+import { StacksCoreRpcClient, getCoreNodeEndpoint } from '../core-rpc/client';
+import { assertNotNullish } from '../helpers';
+import * as compose from 'docker-compose';
+import * as path from 'path';
+import Docker = require('dockerode');
+
+const docker = new Docker();
+
+const sender1 = {
+  address: 'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6',
+  privateKey: 'cb3df38053d132895220b9ce471f6b676db5b9bf0b4adefb55f2118ece2478df01',
+};
+
+const recipientAdd1 = 'ST11NJTTKGVT6D1HY4NJRVQWMQM7TVAR091EJ8P2Y';
+
+const HOST = 'localhost';
+const PORT = 20443;
+const URL = `http://${HOST}:${PORT}`;
+
+const stacksNetwork = GetStacksTestnetNetwork();
+
+const isContainerRunning = async (name: string): Promise<boolean> =>
+  new Promise((resolve, reject): void => {
+    docker.listContainers((err: any, containers: any): void => {
+      if (err) {
+        reject(err);
+      }
+
+      const running = (containers || []).filter((container: any): boolean =>
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+        container.Names.includes(name)
+      );
+
+      resolve(running.length > 0);
+    });
+  });
+
+describe('Rosetta API', () => {
+  let db: PgDataStore;
+  let client: PoolClient;
+  let eventServer: Server;
+  let api: ApiServer;
+  let rosettaOutput: any;
+
+  beforeAll(async () => {
+    process.env.PG_DATABASE = 'postgres';
+    await cycleMigrations();
+    db = await PgDataStore.connect();
+    client = await db.pool.connect();
+    eventServer = await startEventServer({ datastore: db, chainId: ChainID.Testnet });
+    api = await startApiServer({ datastore: db, chainId: ChainID.Testnet });
+
+    // remove previous outputs if any
+    fs.rmdirSync('rosetta-output-construction', { recursive: true });
+
+    // build rosetta-cli container
+    await compose.buildOne('rosetta-cli', {
+      cwd: path.join(__dirname, '../../'),
+      log: true,
+      composeOptions: [
+        '-f',
+        'docker-compose.dev.rosetta-cli.yml',
+        '--env-file',
+        'env.construction',
+      ],
+    });
+    // start cli container
+    void compose.upOne('rosetta-cli', {
+      cwd: path.join(__dirname, '../../'),
+      log: true,
+      composeOptions: [
+        '-f',
+        'docker-compose.dev.rosetta-cli.yml',
+        '--env-file',
+        'env.construction',
+      ],
+      commandOptions: ['--abort-on-container-exit'],
+    });
+
+    await waitForBlock(api);
+    await sleep(10000);
+    await transferStx(recipientAdd1, 1000000000, sender1.privateKey, api);
+    await sleep(15000);
+    await transferStx(recipientAdd1, 1000000000, sender1.privateKey, api);
+    await sleep(15000);
+    await transferStx(recipientAdd1, 1000000000, sender1.privateKey, api);
+    await sleep(15000);
+    await transferStx(recipientAdd1, 1000000000, sender1.privateKey, api);
+    await sleep(15000);
+    await transferStx(recipientAdd1, 1000000000, sender1.privateKey, api);
+    await sleep(15000);
```

Comment:
I think you could use the `addressUpdate` datastore event to wait for these transactions instead of sleeping, otherwise the test will take a minimum of 85 seconds to run.

Reply:
Sorry, this PR was supposed to be in the draft for now.  I'm still working on it, I have just copied the code from the original branch and fixed the bugs. I'm still working on it adding Github actions.

Reply:
Sorry I missed the draft status, all good 👍 let me know when it's ready for review